plugin/Form/FormAddressCreator.php

```diff
@@ -236,4 +242,128 @@ private function getUserNameById(string $userId): string
 			return '';
 		}
 	}
+
+	/**
+	 * @param FormData $pFormData
+	 * @param int $addressId
+	 * @return string
+	 * @throws ApiClientException
+	 * @throws UnknownFieldException
+	 * @throws DependencyException
+	 * @throws NotFoundException
+	 */
+	private function handleLogicMessageDuplicateAddressData(FormData $pFormData, int $addressId): string
+	{
+		$requestParams = $this->getAddressDataForApiCall($pFormData);
+		if (isset($requestParams['gdprcheckbox']) && $requestParams['gdprcheckbox']){
+			$requestParams['DSGVOStatus'] = "speicherungzugestimmt";
+		}
+		unset($requestParams['gdprcheckbox']);
+		if (key_exists('newsletter', $requestParams)) {
+			unset($requestParams['newsletter']);
+		}
```

Comment:
newsletter and dsgvo is already set in createOrCompleteAddress()
Is it necessary to overwrite it here?

Reply:
Yes, it is necessary here.
We overwrite to get $requestParams for the "data" Parameter when calling the "read Addresses" API in the "getDataAddressById" method, not newsletter and dsgvo is already set in createOrCompleteAddress()
Based on this document: https://apidoc.onoffice.de/actions/datensatz-lesen/adressen/